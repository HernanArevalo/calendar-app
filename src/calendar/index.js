
export * from './pages/CalendarPage'

export * from './components/CalendarEvent'
export * from './components/NavBar'
export * from './components/CalendarModal'