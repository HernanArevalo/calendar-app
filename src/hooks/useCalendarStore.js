import { useDispatch, useSelector } from "react-redux"


export const useCalendarStore = () => {
  const dispatch = useDispatch();
  
  const { events, activeEvent } = useSelector(state => state.calendar)


  return {
    //* Properties
    events,
    activeEvent,

    //* Methods

  }
}
