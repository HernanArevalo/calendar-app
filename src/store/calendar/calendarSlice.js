import { createSlice } from '@reduxjs/toolkit';
import { addHours } from 'date-fns';

const tempEvent = 
    {
        _id: new Date().getTime(),
        title: 'Cumple',
        notes: 'Comprar pastel',
        start: new Date(),
        end: addHours( new Date(), 2 ),
        bgColor: '#fafafa',
        user: {
          _id: '123',
          name: 'Fernando'
        }
    };


export const calendarSlice = createSlice({
    name: 'calendar',
    initialState: {
        events: [
            tempEvent
        ],
        activeEvent: null
    },
    reducers: {
        onSetActiveEvent: (state, { payload }) => {
            state.activeEvent = payload;
        }
    }
});


// Action creators are generated for each case reducer function
export const { onSetActiveEvent } = calendarSlice.actions;