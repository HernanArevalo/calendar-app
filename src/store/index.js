export * from './store'

export * from './auth/authSlice'
export * from './calendar/calendarSlice'
export * from './ui/uiSlice'
