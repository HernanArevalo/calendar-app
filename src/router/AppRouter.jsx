import { Route, Routes } from 'react-router-dom'
import { LoginPage } from '../auth'
import { CalendarPage } from '../calendar'

export const AppRouter = () => {

    const authStatus = 'not-authenticated'

  return (
    <Routes>
        {/* TODO: {
            (authStatus === 'not-authenticated')?
             */}
        <Route path='/auth/*' element={ <LoginPage /> }/>
        <Route path='/*' element={ <CalendarPage /> }/>
    </Routes>
  )
}
