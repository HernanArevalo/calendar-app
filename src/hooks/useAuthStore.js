import { useDispatch, useSelector } from "react-redux"
import { calendarApi } from "../api";
import { onChecking, onLogin } from "../store";

export const useAuthStore = () => {


    const { status, user, errorMessage } = useSelector(state => state.auth);
    const dispatch = useDispatch();

    const startLogin = async({ email, password }) => {
        dispatch( onChecking() );

        try{

            const { data } = await calendarApi.post('/auth', {email, password});

            localStorage.setItem('token', data.token);
            localStorage.setItem('token-init-date', new Date().getTime() );


            dispatch( onLogin({ name: data.name, uid: data.uid }) );

        } catch (error){    
            console.log({ error });
        }


    }


    return {
        //* Properties
        errorMessage,
        status, 
        user, 
        //* Methods
        startLogin,
    }
}