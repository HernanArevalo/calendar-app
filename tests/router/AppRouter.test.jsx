import { render, screen } from "@testing-library/react";
import { useAuthStore } from "../../src/hooks";
import { AppRouter } from "../../src/router/AppRouter";
import { MemoryRouter } from "react-router-dom";
import { CalendarPage } from "../../src/calendar";

jest.mock('../../src/hooks/useAuthStore');

jest.mock('../../src/calendar', () => ({
    CalendarPage: () => <h1>CalendarPage</h1>
}))


describe('Pruebas en AppRouter', () => {

    const mockcheckAuthToken = jest.fn();
    beforeEach( () => jest.clearAllMocks() );

    test('Debe mostrar la pantalla de carga y llamar checkAuthToken', () => {

        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockcheckAuthToken
        });

        render( <AppRouter /> );

        expect( screen.getByText('Cargando')).toBeTruthy();
        expect( mockcheckAuthToken ).toHaveBeenCalled();


    });

    test('Debe mostrar el login si no estÃ¡ autenticado', () => {

        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockcheckAuthToken
        });

        const container = render( 
        <MemoryRouter initialEntries={['/asd/']}>
            <AppRouter /> 
        </MemoryRouter>
        );

        expect( screen.getByText('Ingreso') ).toBeTruthy();
        expect( container ).toMatchSnapshot();

    });

    test('Debe mostrar el calendario si estamos autenticados', () => {

        useAuthStore.mockReturnValue({
            status: 'authenticated',
            checkAuthToken: mockcheckAuthToken
        });

        render( 
        <MemoryRouter initialEntries={['/asd/']}>
            <AppRouter /> 
        </MemoryRouter>
        );

        expect( screen.getByText('CalendarPage') ).toBeTruthy();

        screen.debug();

    })


})