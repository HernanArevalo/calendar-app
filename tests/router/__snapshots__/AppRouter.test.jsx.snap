// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe mostrar el login si no está autenticado 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container login-container"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6 login-form-1"
          >
            <h3>
              Ingreso
            </h3>
            <form>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="loginEmail"
                  placeholder="Correo"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="loginPassword"
                  placeholder="Contraseña"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="d-grid gap-2"
              >
                <input
                  class="btnSubmit"
                  type="submit"
                  value="Login"
                />
              </div>
            </form>
          </div>
          <div
            class="col-md-6 login-form-2"
          >
            <h3>
              Registro
            </h3>
            <form>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="registerName"
                  placeholder="Nombre"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="registerEmail"
                  placeholder="Correo"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="registerPassword"
                  placeholder="Contraseña"
                  required=""
                  type="password"
                  value=""
                />
              </div>
              <div
                class="form-group mb-2"
              >
                <input
                  class="form-control"
                  name="registerPassword2"
                  placeholder="Repita la contraseña"
                  required=""
                  type="password"
                  value=""
                />
              </div>
              <div
                class="form-group mb-2"
              >
                <input
                  class="btnSubmit"
                  type="submit"
                  value="Crear cuenta"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container login-container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6 login-form-1"
        >
          <h3>
            Ingreso
          </h3>
          <form>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="loginEmail"
                placeholder="Correo"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="loginPassword"
                placeholder="Contraseña"
                type="password"
                value=""
              />
            </div>
            <div
              class="d-grid gap-2"
            >
              <input
                class="btnSubmit"
                type="submit"
                value="Login"
              />
            </div>
          </form>
        </div>
        <div
          class="col-md-6 login-form-2"
        >
          <h3>
            Registro
          </h3>
          <form>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="registerName"
                placeholder="Nombre"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="registerEmail"
                placeholder="Correo"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="registerPassword"
                placeholder="Contraseña"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                class="form-control"
                name="registerPassword2"
                placeholder="Repita la contraseña"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                class="btnSubmit"
                type="submit"
                value="Crear cuenta"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
